### PR TITLE
Refactor cli

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: refactor cli

### CODE REVIEW

File: lib/src/reporters/models/console_reporter.dart [hunk 1]
```diff
@@ -6,6 +6,9 @@ import 'reporter.dart';
 
 /// Plain terminal reporter
 abstract class ConsoleReporter extends Reporter {
+  static const String id = 'console';
+  static const String alternativeId = 'console-verbose';
```
Comment (dkrutskikh) [inline lib/src/reporters/models/console_reporter.dart:10]: @incendial i think `verboseId` looks better

File: lib/src/cli/arguments_builder/models/parsed_arguments.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import '../../../analyzers/lint_analyzer/metrics/models/metric_value_level.dart';
+
+class ParsedArguments {
```
Comment (dkrutskikh) [inline lib/src/cli/arguments_builder/models/parsed_arguments.dart:3]: @incendial please add `@immutable` annotation

File: lib/src/cli/arguments_builder/models/parsed_arguments.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  final String rootFolder;
+  final String reporterName;
+  final String reportFolder;
+  final MetricValueLevel? metricValueLevel;
```
Comment (dkrutskikh) [inline lib/src/cli/arguments_builder/models/parsed_arguments.dart:7]: @incendial `metricValueLevel` looks not clear, try to rename to `maximumAllowedLevel`

File: lib/src/cli/arguments_builder/models/parsed_arguments.dart [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  final Iterable<String> folders;
+  final String exclude;
```
Comment (dkrutskikh) [inline lib/src/cli/arguments_builder/models/parsed_arguments.dart:9]: @incendial maybe `excludePath` or `exludeFolder`

File: lib/src/cli/arguments_builder/arguments_builder.dart [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import '../../analyzers/lint_analyzer/metrics/metrics_factory.dart';
+import '../../analyzers/lint_analyzer/metrics/models/metric_value_level.dart';
+import 'arguments_parser.dart';
+import 'arguments_validation.dart';
+import 'exceptions/arguments_validation_exceptions.dart';
+import 'models/flag_names.dart';
+import 'models/parsed_arguments.dart';
+
+// ignore: avoid_classes_with_only_static_members
```
Comment (dkrutskikh) [inline lib/src/cli/arguments_builder/arguments_builder.dart:13]: @incendial what's about rewrite this class without static methods?

File: lib/src/analyzers/lint_analyzer/metrics/metrics_factory.dart [hunk 6]
```diff
@@ -23,7 +23,7 @@ final _implementedMetrics = <String, Metric Function(Map<String, Object>)>{
   WeightOfClassMetric.metricId: (config) => WeightOfClassMetric(config: config),
 };
 
-Iterable<Metric> metrics({
+Iterable<Metric> getMetrics({
```
Comment (dkrutskikh) [inline lib/src/analyzers/lint_analyzer/metrics/metrics_factory.dart:26]: @incendial please rename `getRulesById` to `getRules` and `getPatternsById` too

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: lib/src/cli/arguments_builder/arguments_builder.dart (hunk 5 above)
Comment (incendial) [reply lib/src/cli/arguments_builder/arguments_builder.dart:13]: But why?

File: lib/src/cli/arguments_builder/models/parsed_arguments.dart (hunk 2 above)
Comment (incendial) [reply lib/src/cli/arguments_builder/models/parsed_arguments.dart:6]: Done.

File: lib/src/cli/arguments_builder/models/parsed_arguments.dart (hunk 3 above)
Comment (incendial) [reply lib/src/cli/arguments_builder/models/parsed_arguments.dart]: Done.

File: lib/src/cli/arguments_builder/models/parsed_arguments.dart (hunk 4 above)
Comment (incendial) [reply lib/src/cli/arguments_builder/models/parsed_arguments.dart]: Done.

File: lib/src/analyzers/lint_analyzer/metrics/metrics_factory.dart (hunk 6 above)
Comment (dkrutskikh) [reply lib/src/analyzers/lint_analyzer/metrics/metrics_factory.dart:26]: Ok

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**